R/check_forecasts.R

```diff
@@ -0,0 +1,267 @@
+
+    } else if (!check[["target_type"]] == "binary") {
+      errors <- c(errors,
+                  "This forecast does not seem to be for a binary prediction target, so we need a column called quantile or sample")
+    }
+  }
+
+  # check whether there is more than one prediction for the same target, i.e.
+  # the length of prediction is greater 1 for a sample / quantile for
+  # a single forecast
+  type <- c("sample", "quantile")[c("sample", "quantile") %in% colnames(data)]
+  data[, InternalDuplicateCheck := .N, by = c(obs_unit, type)]
+
+  if (any(data$duplicatecheck) > 1) {
+    errors <- c(errors,
+                paste("There are instances with more than one forecast for the same target.",
+                      "This can't be right and needs to be resolved. Maybe you need to check",
+                      "the unit of a single forecast and add missing columns?"))
+    check[["duplicate_forecasts"]] <- data[InternalDuplicateCheck > 1]
+  }
+  data[, InternalDuplicateCheck := NULL]
+
+  # check whether there is a model column present. And if not, state what that means
+  if (!("model" %in% colnames(data))) {
+    msg <- c(msg,
+             paste("There is no column called `model` in the data.",
+                   "scoringutils therefore thinks that all forecasts come from the same model"))
+    data[, model := "Unspecified model"]
+
+  }
+
+  # some checks whether there are the same number of quantiles, samples
+  data[, InternalNumCheck := length(prediction), by = obs_unit]
+  n <- unique(data$InternalNumCheck)
+  if (length(n) > 1) {
+    warnings <- c(warnings,
+                  paste0("Some forecasts have different numbers of rows (e.g. quantiles or samples). ",
+                         "scoringutils found: ", paste(n, collapse = ", "),
+                         ". This is not necessarily a problem, but make sure this is intended."))
+  }
+  check[["rows_per_forecast"]] <-
+    data[, .(rows_per_forecast = unique(InternalNumCheck)), by = model]
+  data[, InternalNumCheck := NULL]
+
+  # get available unique values per model for the different columns
+  cols <- obs_unit[!(obs_unit == "model")]
+  check[["unique_values"]] <-
+    data[, lapply(.SD, FUN = function(x) length(unique(x))), by = "model"]
+
+  check[["messages"]] <- unlist(msg)
+  check[["warnings"]] <- unlist(warnings)
+  check[["errors"]] <- unlist(errors)
+
+  attr(check, "class") <- "scoringutils_check"
+
+  return(check)
+}
+
+
+#' @title Print output from `check_forecasts()`
+#'
+#' @description Helper function that prints the output generated by
+#' [check_forecasts()]
+#'
+#' @param x An object of class 'scoringutils_check' as produced by
+#' [check_forecasts()]
+#' @param ... additional arguments (not used here)
+#'
+#' @return NULL
+#' @method print scoringutils_check
```

Comment:
```suggestion
```

Not necessary in recent versions of roxygen AFAIK

Reply:
hm. unsure. In the beginning I had the problem that the method did not work, but maybe exporting it is enough.